uc-crux-llvm/src/UCCrux/LLVM/Constraints.hs

```diff
@@ -479,6 +474,16 @@ addConstraint modCtx argTypes constraints =
     NewShapeConstraint (SomeInSelector SelectClobbered {}) _ ->
       unimplemented "addConstraint" ClobberConstraints
   where
+    fromMaybeL :: a -> Lens.Lens' (Maybe a) a
+    fromMaybeL def = lens (fromMaybe def) (\_old new -> Just new)
+
+    atDefault ::
+      Lens.At c =>
+      Lens.Index c ->
+      Lens.IxValue c ->
+      Lens.Lens' c (Lens.IxValue c)
+    atDefault idx def = at idx . fromMaybeL def
```

Comment:
It isn't technically necessary, but it might be nice to introduce `a` and `c` with explicit `forall`s to make it clear that these are not captured variables (or float to the top level)